doc/UsersGuide/source/packagemanager.rst

```diff
@@ -1,70 +1,143 @@
+and that you drag-and-drop components from.
+
+This dependency is automatically marked in your package by adding a `uses annotation
+<https://specification.modelica.org/maint/3.5/annotations.html#version-handling>`_ at the top level. For example, if you
+drag and drop components from MSL 4.0.0 into models of your package, the ``annotation(uses(Modelica(version="4.0.0")));``
+will be added automatically to it. This information allows OpenModelica to automatically load all the libraries
+that are required to compile the models in your own package next time you (or someone else, possibly on a different
+computer) loads your package, provided they are installed in places on the computer's file system where OpenModelica
+can find them.
+
+The default place where OpenModelica looks for packages is the so-called 
+`MODELICAPATH <https://specification.modelica.org/maint/3.5/packages.html#the-modelica-library-path-modelicapath>`_.
+You can check where it is by typing ``getModelicaPath()`` in the Interactive Environment. Installed
+read-only libraries are placed by default in the MODELICAPATH.
+
+When a new version of certain package comes out, `conversion annotations
+<https://specification.modelica.org/maint/3.5/annotations.html#version-handling>`_ in it declare whether your models using
+a certain older version of it can be used as they are with the new one, which is then 100% backwards compatible, or whether
+they need to be upgraded by running a conversion script, provided with the new version of the package. 
+
+If you install a new version of a library which is 100% backwards compatible with the previous ones, all your models that
+used the old one will automatically use the new one, which may contain bug-fixes, without the need of any further action.
+
+If the new version is not backwards compatible, instead, you will need to create a new version of
+your library that uses it, by running the provided conversion scripts. 
+
+OpenModelica has a package manager that can be used to install and update libraries on your computer, and is able to run
+conversion scripts. Keep in mind there are three stages in package usage: *available* packages are stored on the OSMC
+servers; *installed* packages are stored in the MODELICAPATH of your computer; *loaded* packages are loaded in memory
+in an active OMC session, either via the Interactive Environment, or via the OMEdit GUI, where they are shown in the
+Libraries Browser. When you load a package, OpenModelica tries to load the best possible installed versions of all
+the dependencies declared in the uses annotation.
+
+The Package Manager
+---------------
+
+The Open Source Modelica Consortium (OSMC) maintains a collection of publicly available, open-source Modelica packages
+on its servers. They are routinely tested with past released versions of OpenModelica, as well as with the current development
```

Comment:
I would use and/or instead of as well (since we sometimes only test master, sometimes only latest released version as requested by developers)

Reply:
Here I was referring to past and development versions _of OMC_, not of the libraries.